apps/src/templates/teacherDashboard/TeacherSectionSelector.jsx

```diff
@@ -76,6 +78,10 @@ export default class TeacherSectionSelector extends Component {
 
   chooseMenuItem = section => {
     this.props.onChangeSection(section.id);
+    updateQueryParam('section_id', section.id);
+    // If we have a user_id when we switch sections we should get rid of it
+    updateQueryParam('user_id', undefined);
+    reload();
```

Comment:
This approach seems less than ideal because I would expect to be able to change the section and see accurate assignment information pertaining to that section without reloading the page, but it's how we currently do it for the old `SectionSelector` https://github.com/code-dot-org/code-dot-org/blob/3437c95f8ea5e0cd213a9c6295030f2f28792fc3/apps/src/code-studio/components/progress/SectionSelector.jsx#L48

Questions for reviewers: 
1.) Is this behavior worth changing? And worth changing now?
2.) If so, is the best way to avoid the reload, connecting `CourseOverview` to a Redux store so it can listen for changes to  selectedSectionId?

Reply:
Good question! I would suggest you do the reload outside of the new component, so that each page can decide separately whether to reload or to try and update all its info another way. initially they may all have to reload, but this way we'll be slightly better positioned to change that in the future.

Reply:
I found a way through my other work on this feature to avoid the reload. Now the course and unit overview pages all update the query param and content as expected when the dropdown changes.